File: frontend/src/types/job.ts
export interface Job {
    id: number;
    title: string;
    company: string;
    location: string;
    department: string;
    employment_type: string;
    url: string;
    description?: string;
    status: "ACTIVE" | "INACTIVE";
    posted_date: string;
    match_percentage: number;
    matched_skills: string[];
    missing_skills: string[];
}

export interface JobsResponse {
    count: number;
    total_count: number;
    page: number;
    page_size: number;
    total_pages: number;
    has_next: boolean;
    has_previous: boolean;
    jobs: Job[];
}

File: app/models/job.py
from datetime import datetime,UTC

from enum import Enum

from sqlalchemy import DateTime, Enum as SqlEnum, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.database import Base


class JobStatus(str, Enum):
    ACTIVE = "ACTIVE"
    INACTIVE = "INACTIVE"


class Job(Base):
    __tablename__ = "jobs"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)

    greenhouse_job_id: Mapped[str] = mapped_column(
        String(100),
        unique=True,
        nullable=False
    )

    title: Mapped[str] = mapped_column(
        String(255),
        nullable=False
    )

    company: Mapped[str] = mapped_column(
        String(255),
        nullable=False
    )

    location: Mapped[str] = mapped_column(
        String(255),
        nullable=False
    )

    department: Mapped[str] = mapped_column(
        String(255),
        nullable=True
    )

    employment_type: Mapped[str] = mapped_column(
        String(100),
        nullable=True
    )

    description: Mapped[str] = mapped_column(
        Text,
        nullable=False
    )

    url: Mapped[str] = mapped_column(
        String(500),
        nullable=False
    )

    status: Mapped[JobStatus] = mapped_column(
        SqlEnum(JobStatus, native_enum=False, length=8),
        default=JobStatus.ACTIVE,
        nullable=False,
        index=True,
    )

    inactive_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True),
        nullable=True,
    )

    created_at: Mapped[datetime] = mapped_column(
    DateTime(timezone=True),
    default=lambda: datetime.now(UTC)
)
